### PR TITLE
Move away from deprecated Teams API Endpoints

### PR DESCRIPTION
**Pull request by pzmi-f3**
# Background

On March 15, 2022 at 14:00 UTC GitHub will deprecate an endpoint under path `/teams/:team_id`
Source: https://github.blog/changelog/2022-02-22-sunset-notice-deprecated-teams-api-endpoints/

# Changes

To use a new endpoint this PR upgrades `google/go-github` module to version v42.0.0.
This is a minimal set of changes to mitigate effects of the deprecation:

- replace call to `githubClient.Teams.ListTeamMembers` with `githubClient.Teams.ListTeamMembersByID`
- add a call to `githubClient.Organizations.Get` - to retrieve an organisation ID, which is required `ListTeamMembersByID`, from only it's name
- adjust test accordingly

# Proof of work

1. I deployed the change in our internal organisation and verified it worked during the brownout period (March 1, 2022 for 12 hours starting at 14:00 UTC).
2. All tests are passing.

### CODE REVIEW

File: internal/api/github/client.go [hunk 1]
```diff
@@ -264,21 +265,36 @@ func (c *Client) GetTeamMembers(ctx context.Context, teams []*github.Team, organ
...
+		org, resorg, err := c.githubClient.Organizations.Get(ctx, organisation)
+		if err != nil {
+			fn()
+			return nil, fmt.Errorf("error getting an organisation %q: %w", organisation, err)
+		}
+		if resorg.StatusCode >= 300 {
+			fn()
+			return nil, fmt.Errorf("error getting an organisation organisation %q (status: %d): %s", organisation, resorg.StatusCode, readAllClose(resorg.Body))
+		}
+		fn()
```
Comment (janakerman-form3) [inline internal/api/github/client.go:277]: I can see you're just copying the existing pattern with calling `fn()` in every branch so I'm happy to merge as is, but a few Qs:
1. Isn't this a prime case for using `defer`?
2. Why do we cancel the context (`fn()`) at all? The places that we do this, we've already exited all the function calls that we've passed `ctxTimeout` to.

**Review by janakerman-form3 — APPROVED** [review]
Approved but with questions!

File: internal/api/github/client.go [hunk 2]
```diff
@@ -48,13 +48,14 @@ func (c *Client) GetConfiguration(ctx context.Context, ownerLogin, repoName stri
 	// Try to download the contents of the configuration file.
 	ctxTimeout, fn := context.WithTimeout(ctx, DefaultGitHubOperationTimeout)
 	defer fn()
-	r, err := c.githubClient.Repositories.DownloadContents(ctxTimeout, ownerLogin, repoName, configuration.ConfigurationFilePath, nil)
+	r, res, err := c.githubClient.Repositories.DownloadContents(ctxTimeout, ownerLogin, repoName, configuration.ConfigurationFilePath, nil)
 	if err != nil {
-		if strings.Contains(err.Error(), "No file named") { // No better way of distinguishing between errors.
+		if strings.Contains(strings.ToLower(err.Error()), "no file named") { // fixme: look at status code instead of body
```
Comment (tomas-deml-form3) [inline internal/api/github/client.go:53]: Shall we submit an issue for that `fixme` or is this still WIP?

Comment (pzmi-f3) [reply internal/api/github/client.go:53]: Good idea, I'll create an issue for this.

File: internal/api/github/client.go (hunk 1 above)
Comment (pzmi-f3) [reply internal/api/github/client.go:277]: I didn't give it too much thought. As you noted, I just copy/pasted existing pattern. I wanted this PR to change as little as possible to fix the issue without making to much mess. I can take a closer look at it next week.

File: internal/api/github/client.go (hunk 2 above)
Comment (pzmi-f3) [reply internal/api/github/client.go:53]: https://github.com/form3tech-oss/github-team-approver/issues/33

**Pull request merged**